.github/workflows/release.yaml

```diff
@@ -0,0 +1,227 @@
+name: CI on Release Tag
+
+# The jobs below will execute any time a tag is pushed to any branch in the repo
+# and when master is updated.
+on: 
+  push:
+    branches:
+      - 'master'
```

Comment:
I think it will result in the creation of a tag named `master`. 

If we check [this Drone run](https://drone-publish.rancher.io/rancher/system-upgrade-controller/214) in the event of pushing to master, only the steps of clone and build are expected. 

I suggest moving this condition ( on.push.branches.master) to the file `.github/workflows/pull-request.yaml`, and modifying that filename and CI name according.